static/src/javascripts/bootstraps/enhanced/media/main.js

```diff
@@ -204,49 +212,55 @@ define([
 
         // Location of this is important.
         events.bindErrorHandler(player);
-
         player.guMediaType = mediaType;
 
-        player.ready(function () {
-            var vol;
-
-            initLoadingSpinner(player);
-            events.bindGlobalEvents(player);
-            upgradeVideoPlayerAccessibility(player);
-            supportedBrowsers(player);
+        playerSetupComplete = new Promise(function (resolve) {
+            player.ready(function () {
+                var vol;
 
-            player.one('playing', function (e) {
-                if (isFlash(e)) {
-                    beacon.counts('video-tech-flash');
-                } else {
-                    beacon.counts('video-tech-html5');
+                events.bindGlobalEvents(player);
+                events.bindContentEvents(player);
+                if (withPreroll) {
+                    events.bindPrerollEvents(player);
                 }
-            });
+                deferToAnalytics(function () {
+                    events.initOmnitureTracking(player);
+                    events.initOphanTracking(player, mediaId);
+                });
 
-            // unglitching the volume on first load
-            vol = player.volume();
-            if (vol) {
-                player.volume(0);
-                player.volume(vol);
-            }
+                initLoadingSpinner(player);
+                upgradeVideoPlayerAccessibility(player);
+                supportedBrowsers(player);
 
-            player.persistvolume({namespace: 'gu.vjs'});
+                player.one('playing', function (e) {
+                    if (isFlash(e)) {
+                        beacon.counts('video-tech-flash');
+                    } else {
+                        beacon.counts('video-tech-html5');
+                    }
+                });
 
-            deferToAnalytics(function () {
+                // unglitching the volume on first load
+                vol = player.volume();
+                if (vol) {
+                    player.volume(0);
+                    player.volume(vol);
+                }
 
-                events.initOmnitureTracking(player);
-                events.initOphanTracking(player, mediaId);
+                player.persistvolume({namespace: 'gu.vjs'});
 
                 // preroll for videos only
                 if (mediaType === 'video') {
-
                     player.fullscreener();
 
-                    if (withPreroll && !blockVideoAds) {
+                    if (showEndSlate && detect.isBreakpoint({ min: 'desktop' })) {
```

Comment:
Yes, we haven't tested it, and we're still uncertain about how ads optimise for slow connections.
We do need to revisit this, perhaps with commercial and another PR.

CC @kenlim